### DISCUSSION TITLE
[Closed] Should the Properties of `Snackbar` be Immutable?

### DISCUSSION THREAD

**Discussion by TheCodeTraveler** [CommunityToolkit/Maui#284] [General]
### Current Situation

Currently, the Properties in `Snackbar` are mutable.

I.e., a dev can change `Snackbar.Text` after initializing `Snackbar`:

```cs
readonly Snackbar _snackbar = new();
readonly DateTime _creationTime = DateTime.Now;

public async void HandleButtonClick(object sender, EventArgs e)
{
    var timeElapsted = DateTime.Now - _creationTime;

    _snackbar.Text = $"It has been {timeElapsted.TotalSeconds} seconds since creation";

    await _snackbar.Show();
}
```

### Problem

However, if a `Snackbar` is currently displayed, changing the value of its properties, e.g. `Snackbar.Text = "New Text";`, does not update the displayed Snackbar; the new text will only be displayed once `Snackbar.Show()` is called again.

This is different from the behavior of `ILabel`, for example; `Label.Text` updates on screen immediately when a dev changes its value.

### Proposal

I propose that we make `Snackbar` properties immutable before we promote `CommunityToolkit.Maui` from preview to stable. 

After we debut the first Stable Release of `CommunityToolkit.Maui`, we can always make `Snackbar` mutable without breaking changes. However, if we release `Snackbar` with mutable properties, changing it to be immutable _will_ be a breaking change.

```cs
/// <inheritdoc/>
public SnackbarOptions VisualOptions { get; init; }

/// <inheritdoc/>
public string Text { get; init; }

/// <inheritdoc/>
public TimeSpan Duration { get; init; }

/// <inheritdoc/>
public Action? Action { get; init; }

/// <inheritdoc/>
public string ActionButtonText { get; init; }

/// <inheritdoc/>
public IView? Anchor { get; init; }

/// <inheritdoc/>
public static bool IsShown { get; private set; }
```

**Comment by pictos** [comment]
@brminnick I believe that we can resolve this discussion already, we're on 🔥

**Reply by TheCodeTraveler** [reply]
Resolved in https://github.com/CommunityToolkit/Maui/issues/238#issuecomment-1032071004

**Comment by CliffAgius** [comment]
Closed as answered.
